src/x11.cc

```diff
@@ -1205,61 +1084,83 @@ void print_desktop_name(struct text_object *obj, char *p,
 }
 
 #ifdef OWN_WINDOW
+enum class x11_strut : size_t {
+  LEFT,
+  RIGHT,
+  TOP,
+  BOTTOM,
+  LEFT_START_Y,
+  LEFT_END_Y,
+  RIGHT_START_Y,
+  RIGHT_END_Y,
+  TOP_START_X,
+  TOP_END_X,
+  BOTTOM_START_X,
+  BOTTOM_END_X,
+  COUNT
+};
+constexpr size_t operator*(x11_strut index) {
+  return static_cast<size_t>(index);
+}
+
 /* reserve window manager space */
-void set_struts(int sidenum) {
-  Atom strut;
-  if ((strut = ATOM(_NET_WM_STRUT)) != None) {
+void set_struts(alignment align) {
+  // Middle and none align don't have least significant bit set.
+  // Ensures either vertical or horizontal axis are start/end
+  if ((*align & 0b0101) == 0) return;
+
+  Atom strut = ATOM(_NET_WM_STRUT);
+  if (strut != None) {
     /* reserve space at left, right, top, bottom */
-    signed long sizes[12] = {0};
+    uint32_t sizes[*x11_strut::COUNT] = {0};
     int i;
 
-    /* define strut depth */
-    switch (sidenum) {
```

Comment:
Here, instead of matching again on intermediate sidenum, we can just match actual alignment values. This is clearer, **and** generates much less assembly (though a lot more verbose).